backend/Application/Api/GraphQL/Import/AccountLookup.cs

```diff
@@ -52,10 +78,32 @@ public AccountLookup(IMemoryCache cache, DatabaseSettings dbSettings, IMetrics m
                 AddToCache(account.Key, account.Result);
                 result.Add(account);
             }
-
-            var nonExistingAccounts = notCached.Except(result.Select(x => x.Key));
-            foreach (var nonExistingAccount in nonExistingAccounts)
+        }
+        
+        var nonExistingAccounts = notCached.Except(result.Select(x => x.Key));
```

Comment:
Consider dividing the `GetAccountIdsFromBaseAddresses` method into sub-methods that correspond to each of the steps you described in the summary